examples/subscribing_to_stream.rs

```diff
@@ -0,0 +1,112 @@
+use eventstore::{
+    Client, Credentials, EventData, ExpectedVersion, Origin, Position, ReadResult, SubEvent,
+    SubscribeToAllOptions, SubscribeToStreamOptions, SubscriptionFilter,
+};
+use futures::TryStreamExt;
+use serde::{Deserialize, Serialize};
+use std::error::Error;
+use uuid::Uuid;
+
+type Result<A> = std::result::Result<A, Box<dyn Error>>;
+
+pub async fn subscribe_to_stream(client: &Client) -> Result<()> {
+    // region subscribe-to-stream
+    client
+        .subscribe_to_stream("some-stream", &Default::default())
+        .await?;
+    // endregion subscribe-to-stream
+
+    // region subscribe-to-stream-from-position
+    let options = SubscribeToStreamOptions::default().origin(Origin::Point(20));
+
+    client.subscribe_to_stream("some-stream", &options).await?;
+    // endregion subscribe-to-stream-from-position
+
+    // region subscribe-to-stream-live
+    let options = SubscribeToStreamOptions::default().origin(Origin::End);
+    client.subscribe_to_stream("some-stream", &options).await?;
+    // endregion subscribe-to-stream-live
+
+    // region subscribe-to-stream-resolving-linktos
+    let options = SubscribeToStreamOptions::default()
+        .origin(Origin::Start)
+        .resolve_links();
+
+    client
+        .subscribe_to_stream("$et-myEventType", &options)
+        .await?;
+    // endregion subscribe-to-stream-resolving-linktos
+
+    // region subscribe-to-stream-subscription-dropped
+    let mut stream = client
+        .subscribe_to_stream("some-stream", &Default::default())
+        .await?;
+
+    while let Some(event) = stream.try_next().await? {
+        if let SubEvent::EventAppeared(event) = event {
+            // Handles the event...
+        }
+    }
+
+    // endregion subscribe-to-stream-subscription-dropped
+
+    Ok(())
+}
+
+pub async fn subscribe_to_all(client: &Client) -> Result<()> {
+    // region subscribe-to-all
+    client.subscribe_to_all(&Default::default()).await;
+    // endregion subscribe-to-all
+
+    // region subscribe-to-all-from-position
+    let options = SubscribeToAllOptions::default().origin(Origin::Point(Position {
```

Comment:
Same comment as above: https://github.com/EventStore/EventStoreDB-Client-Rust/pull/49/files#r572702828.

Reply:
Similar to https://github.com/EventStore/EventStoreDB-Client-Rust/pull/49#discussion_r573642574

Reply:
done